dbms/src/Functions/FunctionsRegexp.h

```diff
@@ -90,18 +93,29 @@ inline int getDefaultFlags()
     return flags;
 }
 
+template <bool need_subpattern>
```

Comment:
How about we let the caller to make sure the input pattern is already a subpattern so we don't need to add this template argument for all related functions?

Reply:
> How about we let the caller to make sure the input pattern is already a subpattern so we don't need to add this template argument for all related functions?

okk